llama_index/node_parser/interface.py

```diff
@@ -60,18 +60,30 @@ def get_nodes_from_documents(
         ) as event:
             nodes = self._parse_nodes(documents, show_progress=show_progress, **kwargs)
 
-            if self.include_metadata:
-                for node in nodes:
-                    if (
-                        node.ref_doc_id is not None
-                        and node.ref_doc_id in doc_id_to_document
-                    ):
+            for i, node in enumerate(nodes):
+                if (
+                    node.ref_doc_id is not None
+                    and node.ref_doc_id in doc_id_to_document
+                ):
+                    ref_doc = doc_id_to_document[node.ref_doc_id]
+                    start_char_idx = ref_doc.text.find(
+                        node.get_content(metadata_mode=MetadataMode.NONE)
```

Comment:
one thing I worry about is duplicates. if ref_doc has dups for node.content, will it cause issues?

Reply:
eh, it will just return the index of the first match

If a document has copies of the exact same text, I feel like this is an ok side-effect?

Reply:
I could make this logic more complicated by keeping tracking of the previous start idx position, but that assumes the nodes are returned in order

Reply:
ic that is fine. just raise my concern. but it could possibly happen in very rare cases